pkg/execute/kubectl/merger.go

```diff
@@ -0,0 +1,119 @@
+package kubectl
+
+import (
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+// EnabledKubectl configuration for executing commands inside cluster
+type EnabledKubectl struct {
+	AllowedKubectlVerb     map[string]struct{}
+	AllowedKubectlResource map[string]struct{}
+
+	DefaultNamespace string
+	RestrictAccess   bool
+}
+
+// Merger provides functionality to merge multiple bindings
+// associated with the kubectl executor.
+type Merger struct {
```

Comment:
We can create a wrapper to cache the calc responses. Example pseudo code:

```go
package kubectl

import (
	"fmt"

	"github.com/kubeshop/botkube/pkg/config"
)

type Merger interface {
	MergeForNamespace(includeBindings []string, forNamespace string) EnabledKubectl
	MergeAllEnabled(includeBindings []string) EnabledKubectl
	IsAtLeastOneEnabled() bool
	MergeAllEnabledVerbs(bindings []string) map[string]struct{}
}

type CachedMerger struct {
	underlying Merger
}

// NewMergerWithCache returns a new Merger instance with caching layer.
func NewMergerWithCache(merger Merger) *CachedMerger {
	return &CachedMerger{
		underlying: merger,
	}
}

func (m *CachedMerger) MergeAllEnabledVerbs(bindings []string) map[string]struct{} {
	const key = "MergeAllEnabledVerbs"
	m.mux[key].Lock()
	defer m.mux[key].UnLock()

	var entryKey = fmt.Sprintf("%s/%s", key, bindings)
	out, found := m.cache[entryKey]
	if !found {
		out = m.underlying.MergeAllEnabledVerbs(bindings)
		m.cache[entryKey] = out
	}

	return out
}

func (m *CachedMerger) MergeForNamespace(includeBindings []string, forNamespace string) EnabledKubectl {
 // TODO
}

func (m *CachedMerger) MergeAllEnabled(includeBindings []string) EnabledKubectl {
 // TODO
}

func (m *CachedMerger) IsAtLeastOneEnabled() bool {
 // TODO
}

```

We can also use some caching Go lib, e.g. https://github.com/allegro/bigcache